pandas/core/indexing.py

```diff
@@ -1550,6 +1551,13 @@ def _setitem_with_indexer(self, indexer, value):
                 val = list(value.values()) if isinstance(value, dict) else value
                 blk = self.obj._mgr.blocks[0]
                 take_split_path = not blk._can_hold_element(val)
+                if isinstance(value, ABCSeries):
+                    take_split_path = not (is_dtype_equal(value.dtype, blk.dtype))
```

Comment:
so this is for e.g. if value is float32 and blk is float64?

Reply:
For example, or if blk is int64 and value is float64 or vice versa